src/Open-SAMP-API/Client/SAMPFunctions.cpp

```diff
@@ -3,57 +3,53 @@
 #include <Shared/PipeMessages.hpp>
 #include <boost/algorithm/string.hpp>
 
-EXPORT int Client::SAMPFunctions::GetServerIP(char *&ip, int max_len)
-{
+int Client::SAMPFunctions::ReadGTACmdArgument(char *option, char *&str, int max_len) {
 	char *szCommandLine = new char[512];
 	ZeroMemory(szCommandLine, 512);
 
-	GTAFunctions::GetGTACommandLine(szCommandLine, 512);
+	if (GTAFunctions::GetGTACommandLine(szCommandLine, 512) == 0)
+		return 0;
```

Comment:
Consider cleaning up ```szCommandLine```. Otherwise, you have a memory leak there. Sorry, that I didn't give you the feedback before you merged the branch.